lua/orgmode/utils/promise.lua

```diff
@@ -288,6 +289,14 @@ function Promise.wait(self, timeout)
     return error(value)
   end
 
+  if not success and code == -1 then
+    return error('promise timeout of ' .. tostring(timeout) .. 'ms reached')
+  elseif not success and code == -2 then
+    return error('promise interrupted')
+  elseif not success then
+    return error('promise failed with unknown reason')
+  end
+
   return value
```

Comment:
Lets reverse the success check to make the conditionals more readable:

```suggestion
  if success then
    return value
  end

  if code == -1 then
    return error('promise timeout of ' .. tostring(timeout) .. 'ms reached')
  elseif code == -2 then
    return error('promise interrupted')
  end

  return error('promise failed with unknown reason')
```